### DISCUSSION TITLE
Refreshed software after 9 months idle, not working right!

### DISCUSSION THREAD

**Discussion by acidkewpie** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#387] [Q&A]
Hi, probably a genuine stupid user question, but I've just brought my Helsinki back to life in line with changing to an overnight electricity tariff. with a 9 month gap I see things have moved on a lot. The device was working fine this morning - outside of the buttons on the web page taking a while to have effect - always been that way, presumed it was normal, esp when multiple instances are open?) I've upgraded to 3.4.1 with various amounts of OTA and USB updates, and both the filesystem and the code appear to be up to date. New UI reports "Firmware version: 2022-10-08, Model: PRE2021"

But it's not working right. I see the pins were changed, so I've got "#define PCB_V1" as opposed to V2 (is it worth updating the connections to this v2 layout for some other reason..?) and now it *looks* like it's all sorted, but when I try and set the temperature on the app, it goes bonkers for a few seconds, pinging back and forth before settling back down to minimum temp again, and then the buttons on the unit stop responding. Also random fun with the timer coming on, and such. Essentially it seems to work normally from boot if I don't use the web interface, if I do, all bets are off! Hopefully these symptoms sound like an obvious root cause!

**Comment by wyliecoyoteuk** [comment]
What version of Pump? egg or square?

Have you tried the hardware test? http://layzspa.local/hwtestinfo.html

**Reply by acidkewpie** [reply]
It's a square. Had no idea that test existed looks super useful, not got a 5v supply to hand immediately though. Will try and sort one.

**Reply by acidkewpie** [reply]
DSP input pin 2 0 errors of 1000
DSP input pin 3 0 errors of 1000
**DSP input pin 4 41 errors of 1000**
CIO input pin 2 0 errors of 1000
CIO input pin 3 0 errors of 1000
CIO input pin 4 0 errors of 1000

Signficiant? It's usually zero, but goes that high sometimes.

**Reply by visualapproach** [reply]
@TheRocketSurgeon It should ideally say zero. It indicates some problem. Corrosion or broken wire maybe. Reseat the chips if you have female headers. And resolder that wire, and perhaps the pins on the same trace towards the esp.
No need to switch to PCB V2 if you have a v1.

**Reply by visualapproach** [reply]
Are you sure it's a pre2021 and not miami2021?

**Reply by acidkewpie** [reply]
Ahem... maybe.... feeling suitably embarrassed now. cheers :)

**Reply by visualapproach** [reply]
> Ahem... maybe.... feeling suitably embarrassed now. cheers :)

Cheers mate! Thank you 🙏🏻

**Comment by Jay-Ti** [comment]
Does the ESP keep working, or does it watchdog reset, does it accept HTTP requests, do the Cio Dsp channels stay alive?

**Reply by acidkewpie** [reply]
The ESP itself seems to think it's working, doesn't crash, just acts mad. I don't know what you mean by the Cio Dsp channels though, was planning on getting a laptop connected to it with platformio and monitoring it there compared to the debug.html console output in case that's any different.

**Reply by Jay-Ti** [reply]
even if it crashes its back within a few seconds. There is a Reset reason and time on the website under Spa config (i think)

**Reply by Jay-Ti** [reply]
Cio and Dsp channels should be shown as active or deadon the bottom of the main page

**Reply by wyliecoyoteuk** [reply]
CIO = control unit DSP = display unit.
If you are getting errors in the hardware test like that it is most likely noise from a poor connection
